sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -39,11 +39,12 @@ object DefaultOptimizer extends Optimizer {
     Batch("Distinct", FixedPoint(100),
       ReplaceDistinctWithAggregate) ::
     Batch("Operator Optimizations", FixedPoint(100),
-      UnionPushdown,
+      UnionPushDown,
       CombineFilters,
       PushPredicateThroughProject,
       PushPredicateThroughGenerate,
       ColumnPruning,
+      LimitPushDown,
```

Comment:
`LimitPushDown` should run after `ColumnPruning`. For something like `Limit(Project(Sort(...)))`, we should try to push down `Project` through `Sort` first.